Security/How to Configure a Service Fabric Managed Cluster with Common Name Certificate.md

```diff
@@ -0,0 +1,259 @@
+This will copy the certificate to each of the nodes into the appropriate certificate store. Before the application can use the certificate, ACL'ing of the certificate will need to be configured. See [How to ACL application certificate private key using ApplicationManifest.xml](./How%20to%20ACL%20application%20certificate%20from%20ApplicationManifest.md).
+
+Use one of the three options below  to add a common name certificate for application connectivity.
+
+### Using Azure Resource Explorer to add common name certificate to nodetype configuration
+
+- Open https://resources.azure.com and navigate to {{subscription id}}/resourceGroups/{{resource group}}/providers/Microsoft.ServiceFabric/managedClusters/{{cluster name}}/nodeTypes/{{nodetype name}}
+- Select 'Read/Write' and 'Edit'
+- Populate provided 'vmSecrets' new element template
+  - id - Azure key vault id. Example: '/subscriptions/{{subscription id}}/resourceGroups/xxxxxxx/providers/Microsoft.KeyVault/vaults/{{vault name}}'
+  - certificateStore - 'My'
+  - certificateUrl - Azure key vault certificate secret. Example: 'https://{{vault name}}.vault.azure.net:443/secrets/{{secret name}}/{{secret}}'
+
+  ```json
+  //"vmSecrets": [
+      {
+        "sourceVault": {
+          "id": "(string)"
+        },
+        "vaultCertificates": [
+          {
+            "certificateUrl": "(string)",
+            "certificateStore": "My"
+          }
+        ]
+      }
+  //],    
+  ```
+
+- 'PUT' to update configuration.
+
+### Using PowerShell to add common name certificate to nodetype configuration
+
+Run the commands below to use PowerShell to update the configuration of the cluster nodetype.
+
+Variables:
+
+- resourceGroupName - Azure resource group that contains the managed cluster.
+- certificateStore - 'My'
+- certificateUrl - Azure key vault certificate secret. Example: 'https://{{vault name}}.vault.azure.net:443/secrets/{{secret name}}/{{secret}}'
+- nodeTypeName - Name of nodetype  / vmss that is being modified.
+- sourceVaultId - Azure key vault id. Example: '/subscriptions/{{subscription id}}/resourceGroups/xxxxxxx/providers/Microsoft.KeyVault/vaults/{{vault name}}'
+
+```powershell
+$resourceGroupName = 'sfcluster'
+$certificateUrl = '' #'https://{{vault name}}.vault.azure.net:443/secrets/{{secret name}}/{{secret}}'
+$certificateStore = 'My'
+$clusterName = $resourceGroupName
+$nodeTypeName = 'nodetype1'
+$sourceVaultId = '' #'/subscriptions/{{subscription id}}/resourceGroups/xxxxxxx/providers/Microsoft.KeyVault/vaults/{{vault name}}'
+Add-AzServiceFabricManagedNodeTypeVMSecret -ResourceGroupName $resourceGroupName `
+   -ClusterName $clusterName `
+   -Name $nodeTypeName `
+   -SourceVaultId $sourceVaultId `
+   -CertificateUrl $certificateUrl `
+   -CertificateStore $certificateStore `
+   -Verbose
+```
+
+### Using ARM template to add common name certificate to nodetype configuration
```

Comment:
similar to previous comment, list ARM examples first

Reply:
updated